circle.yml

```diff
@@ -1,37 +1,89 @@
-defaults: &defaults
-  steps:
-    - run:
-        name: Ensure npm is at least at version 3
-        command: |
-          if [[ "$(npm --version)" = [12].* ]]; then
-            npm install --global npm@3
-          fi
-    - checkout
-    - run:
-        name: Install and test
-        command: |
-          npm install --loglevel warn
-          npm test --loglevel warn
-
 version: 2
+
+buildSteps: &buildSteps
+  - run:
+      name: Ensure npm is at least at version 3
+      command: |
+        if [[ "$(npm --version)" = [12].* ]]; then
+          npm install --global npm@3
+        fi
+  - checkout
+  - run:
+      name: Install
+      command: npm install --loglevel warn
+  - run:
+      name: Test
+      command: npm test --loglevel warn
+  - persist_to_workspace:
+      # Persist all job output, so we can (potentially) use it for deploys
+      root: ../
+      paths:
+        - ./node-*
+
 jobs:
-  node4:
+  node-4:
     docker:
-      - image: library/node:4
-    <<: *defaults
-  node6:
+      - image: circleci/node:4
+    working_directory: ~/node-4
+    steps: *buildSteps
+
+  node-6:
     docker:
-      - image: library/node:6
-    <<: *defaults
-  node8:
+      - image: circleci/node:6
+    working_directory: ~/node-6
+    steps: *buildSteps
+
+  node-8:
+    docker:
+      - image: circleci/node:8
+    working_directory: ~/node-8
+    steps: *buildSteps
+
+  deploy:
+    # For this to work NPM_TOKEN must be set for the account used for publishing
     docker:
-      - image: library/node:8
-    <<: *defaults
+      - image: circleci/node:8
+    steps:
+      - attach_workspace:
+          at: $CIRCLE_WORKING_DIRECTORY
+      - run:
+          name: Login
+          command: |
+            echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
+            npm whoami
+      - deploy:
+          name: Deploy
+          command: npm publish
+          # Output used for publish is from node 8 build
```

Comment:
If publishing a node 8 build, should [`package-lock.json`](https://docs.npmjs.com/files/package-lock.json) be included in the repo?

Reply:
package-lock.json is ignored when publishing, so I don't think it matters either way really